### PR TITLE
[Rust] entry-api - replace about.md file with concept files

### PR DESCRIPTION
**Pull request by ErikSchierboom**
In [this issue](https://github.com/exercism/v3/issues/2293) we're describing an evolution of the specification of this repo.
These changes include, amongst others, replacing the contents of the `after.md` document with individual concept documents. There are two documents per concept:

1. An `about.md` file containing the description of the concept
1. A `links.json` file containing a set of useful links related to the concept. 

This looks as follows in the repo:

```
<track>
├── concepts
│   ├── <concept>
│   │   ├── about.md
│   │   └── links.json
│   └── ...
```

This PR applies this change to this exercise by: 
                
1. Copying the existing `after.md` file's contents to the `about.md` file of each concept listed in the exercise's `concepts` section in the `config.json` file.
1. Extract the links from the `after.md` file and put them into the `links.json` file

Before merging this PR, please check that the contents of the concept documents makes sense.
This is especially true when the exercise unlocks multiple concepts, as then the concept documents should be updated to only contain the information relevant to that concept.

Note: to make reviewing easier, the PR has been split into three separate commits. Please squash this PR upon merging.

### CODE REVIEW

File: languages/rust/concepts/entry-api/links.json [hunk 1]
```diff
@@ -1 +1,10 @@
-[]
+[
+  {
+    "url": "https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html#method.insert",
+    "description": "method"
```
Comment (coriolinus) [inline languages/rust/concepts/entry-api/links.json:4]:
```suggestion
    "description": "`Entry::insert`"
```

File: languages/rust/concepts/entry-api/links.json [hunk 2]
```diff
@@ -1 +1,10 @@
...
+  },
+  {
+    "url": "https://doc.rust-lang.org/std/collections/hash_map/enum.Entry.html#method.key",
+    "description": "method"
```
Comment (coriolinus) [inline languages/rust/concepts/entry-api/links.json:8]:
```suggestion
    "description": "`Entry::key`"
```

File: languages/rust/concepts/entry-api/about.md [hunk 3]
```diff
@@ -1 +1,15 @@
-TODO: add information on entry-api concept
+The entry API is often the most idiomatic way to manipulate the contents of a map structure. It's especially useful when you need to alter multiple entries in a map structure.
+
+This exercise covered one particular pattern that the entry API is commonly used for, namely using `HashMap`s to store the frequency of elements.
```
Comment (coriolinus) [inline languages/rust/concepts/entry-api/about.md:3]: If this is general documentation, it shouldn't refer to "this exercise", or the work done in that exercise. Instead, we should just walk about various methods to accomplish the goal.

```rust
*map.entry(key).or_default() += 1;
```

**Comment by coriolinus** [comment]
@exercism/rust as I've rewritten the concept documentation, someone else should now review this.

**Review by petertseng — APPROVED** [review]

**Comment by coriolinus** [comment]
CI failures are irrelevant to this PR; merging anyway.

**Pull request merged**